### PR TITLE
Init

### PR DESCRIPTION
**Pull request by DavyCats**

### CODE REVIEW

**Comment by ffinfo** [comment]
This depends on https://github.com/biowdl/expression-quantification/pull/2

**Comment by ffinfo** [comment]
@DavyCats can this continue again? The scatter issues should be resolved now

**Comment by DavyCats** [comment]
@ffinfo Yes, there is still some call-caching and memory stuff that needs to be fixed.

File: readgroup.wdl [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    output {
+        File cleanR1 = qc.read1afterQC
+        File? cleanR2 = qc.read2afterQC
+        String starRGline = rgLine.rgLine
+    }
+}
+
+
+task makeStarRGline {
```
Comment (rhpvorderman) [inline readgroup.wdl]: Maybe nice to put in the tasks directory? Looks like something that can be reused in other pipelines.

File: sample.wdl [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                    libraryId = lib,
+                    ref_fasta = ref_fasta,
+                    ref_dict = ref_dict,
+                    ref_fasta_index = ref_fasta_index
+            }
+        }
+    }
+
+    # merge library (mdup) bams into one (for counting)
+    call samtools.Merge as mergeLibraries {
```
Comment (rhpvorderman) [inline sample.wdl]: If there is only one Library, this merge step should not be run to save time.

File: sample.wdl [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    if (multiple_bams) {
+        call samtools.Index as mergedIndex {
+            input:
+                bamFilePath = mergeLibraries.outputBam,
+                bamIndexPath = mergeLibraries.outputBam + ".bai"
+        }
+    }
+
+    if (! multiple_bams) {
+        call common.createLink as linkIndex {
```
Comment (rhpvorderman) [inline sample.wdl]: This task is redundant. See comment below

File: sample.wdl [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            ref_fasta_index = ref_fasta_index,
+            bamFiles = select_all(library.preprocessBamFile),
+            bamIndexes = select_all(library.preprocessBamIndexFile),
+            gvcf_basename = sampleDir + "/" + sampleId + ".g"
+    }
+
+    output {
+        String sampleName = sampleId
+        File bam = mergeLibraries.outputBam
+        File bai = select_first([if multiple_bams then mergedIndex.indexFile else linkIndex.link])
```
Comment (rhpvorderman) [inline sample.wdl]: why not use `else select_first(library.bamIndexFile)` instead of the link stuff?

**Pull request merged**